### PR TITLE
feature: 420 add /management/v2/contractagreements/request and GET  /…

### PR DESCRIPTION
**Pull request by ds-lcapellino**
…management/v2/contractagreements/{contractAgreementId}/negotiation to irs-edc-client lib

### CODE REVIEW

File: irs-edc-client/src/main/java/org/eclipse/tractusx/irs/edc/client/contract/model/exception/ContractAgreementException.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ********************************************************************************/
+package org.eclipse.tractusx.irs.edc.client.contract.model.exception;
+
+/**
+ * ContractAgreementException used for failures in
+ * {@link org.eclipse.tractusx.irs.edc.client.contract.service.EdcContractAgreementService}
+ */
+public class ContractAgreementException extends RuntimeException {
```
Comment (ds-jhartmann) [inline irs-edc-client/src/main/java/org/eclipse/tractusx/irs/edc/client/contract/model/exception/ContractAgreementException.java:26]: We usually use checked Exceptions for this. See e.g. EdcClientException, TransferProcessException.
If possible, please align with that.

File: irs-edc-client/src/main/java/org/eclipse/tractusx/irs/edc/client/contract/service/EdcContractAgreementService.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+                        + "/negotiation", ContractNegotiation.class);
+        return contractNegotiationResponseEntity.getBody();
+    }
+
+    private QuerySpec buildQuerySpec(final String... contractAgreementIds) {
+
+        final List<Criterion> criterionList = Arrays.stream(contractAgreementIds)
+                                                    .map(id -> Criterion.Builder.newInstance()
+                                                                                .operandLeft(
+                                                                                        "https://w3id.org/edc/v0.0.1/ns/assetId")
```
Comment (ds-jhartmann) [inline irs-edc-client/src/main/java/org/eclipse/tractusx/irs/edc/client/contract/service/EdcContractAgreementService.java:77]: Pleas add `"https://w3id.org/edc/v0.0.1/ns/assetId"` as constant

**Review by ds-jhartmann — CHANGES_REQUESTED** [review]
Some minor refactoring suggestions, everything else looks good :)

File: charts/irs-helm/values.yaml [hunk 3]
```diff
@@ -173,10 +173,11 @@ edc:
...
     endpoint:
       data: ""  # <edc-controlplane-endpoint-data>
-      catalog: /v2/catalog/request  # EDC consumer controlplane catalog path
-      contractnegotiation: /v2/contractnegotiations  # EDC consumer controlplane contract negotiation path
-      transferprocess: /v2/transferprocesses  # EDC consumer controlplane transfer process path
+      catalog: /management/v2/catalog/request  # EDC consumer controlplane catalog path
+      contractnegotiation: /management/v2/contractnegotiations  # EDC consumer controlplane contract negotiation path
+      transferprocess: /management/v2/transferprocesses  # EDC consumer controlplane transfer process path
       statesuffix: /state  # Path of the state suffix for contract negotiation and transfer process
+      contractAgreements: /management/v2/contractagreements # EDC consumer controlplane contract agreements path
```
Comment (ds-jhartmann) [inline charts/irs-helm/values.yaml:180]: We have templated the `controlplane.endpoint.data` field to `"http://%s%s" .Release.Name "-tractusx-connector-controlplane:8081/management"` by default, so the `/management` path is included there already.
The name "data" might be a bit misleading here.

**Review by ds-jhartmann — CHANGES_REQUESTED** [review]

File: charts/irs-helm/values.yaml (hunk 3 above)
Comment (ds-jhartmann) [inline charts/irs-helm/values.yaml:180]:
```suggestion
      contractAgreements: /management/v2/contractagreements  # EDC consumer controlplane contract agreements path
```

**Comment by ds-lcapellino** [comment]
Thanks for the feedback @ds-jhartmann. Please see the new changes :)

**Review by ds-jhartmann — APPROVED** [review]
LGTM

**Pull request merged**